owncloud/core
# Make it possible to filter by tags with REPORT method

## Description

Enhanced the REPORT method on the Webdav endpoint and added a
"oc:favorite" filter rule. When set, it will return a flat list of
results filtered with only favorite files.

The web UI was also adjusted to use this REPORT method instead of the
private API endpoint.
## Related Issue

Fixes https://github.com/owncloud/core/issues/23263
## Motivation and Context

This makes it possible for other clients like desktop and mobile to retrieve a flat filtered list of favorite files. The previous API was private and was not supposed to be used, but the REPORT method and the webdav endpoint are public. @davivel @nasli @dragotin @guruz 
## How Has This Been Tested?
1. Favorite a few files in folders and subfolders, share one of them with link (for the share indicator). 
2. Go to the "Favorites" section in the web UI. 
3. Check the network console and see that it's using REPORT to get the results.
4. Check that the displayed results look fine and also contain the share indicator from the previous link share (it's part of the Webdav response)
## Screenshots (if appropriate):
## Types of changes


## Checklist:





Please review @DeepDiver1975 @butonic @VicDeo @guruz @davivel @SergioBertolinSG @nasli
